tdigest/tdigest.proto

```diff
@@ -0,0 +1,23 @@
+syntax = "proto3";
+package tdigest;
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+
+// MergingDigestData contains all fields necessary to generate a MergingDigest.
+// This type should generally just be used when serializing MergingDigest's,
+// and doesn't have much of a purpose on its own.
+message MergingDigestData {
+    // Use values rather than pointers for the Centroid array.  This avoids
+    // a ton of code changes and probably a lot of allocations as well.
+    repeated Centroid main_centroids = 1 [(gogoproto.nullable) = false];
```

Comment:
Could you summarize why nullable = false is safe/appropriate for this use case?

Reply:
My intuition is that, since this is a repeated field, it isn't really reasonable that this field will be "unset".  Specifically for Go generated, code, adding this annotation just makes the generated code create slices of structs, rather than slices of pointers to structs:

Without `[(gogoproto.nullable) = false]`:

```
type MergingDigestData struct {
    MainCentroids []*Centroid `protobuf:"bytes,1,rep,name=main_centroids,json=mainCentroids" json:"main_centroids,omitempty"`
    ...
}
```

With it (current code):

```
type MergingDigestData struct {
    MainCentroids []Centroid `protobuf:"bytes,1,rep,name=main_centroids,json=mainCentroids" json:"main_centroids"`
    ...
}
```

In Go, even if some client manages to send an empty value for `MainCentroids` it should just default to an empty slice.  I suppose the real issue will be if a client sends null values in the slice, which would appear in a server as uninitialized `Centroid` structs.  This feature also seems to break compiling these protobufs to other languages, which is a little annoying but I'm not sure if its necessarily a requirement.

I did this to minimize the size of this diff, as removing this annotation would require extra code to convert between the type of the list of centroids on `MergingDigest` (a list of structs) and what's expected in the serialized struct (a list of pointers).  I'm happy to change this back if it seems worth it!